Cards : fix design + add design callout messages (#2001)
Orange-OpenSource/Orange-Boosted-Bootstrap: site/content/docs/5.3/components/card.md

A **card** is a flexible and extensible content container. It includes options for headers and footers, a wide variety of content, contextual background colors, and powerful display options. If you're familiar with Boosted 3, cards replace our old panels, wells, and thumbnails. Similar functionality to those components is available as modifier classes for cards.

> {{< design-callout-alert info >}}

Reviewer: I can't see what's the recommendation here 🤔

> **On the Orange Design System website**, you'll find the [Cards](https://system.design.orange.com/0c1af118d/p/82eaf1-cards/b/69655e) component guidelines describing how Orange designers want cards to be represented and used.

Reviewer: ```suggestion
**On the Orange Design System website**, you'll find the [Cards component guidelines](https://system.design.orange.com/0c1af118d/p/82eaf1-cards/b/69655e) describing how Orange designers want cards to be represented and used.
```
For better a11y

Author: If I do that here, I will have to it in all the other design callout messages :-)

<br><br>
**In this Boosted cards component description page**, you'll find card variants that can be used as cards as well as some variants that are meant to be used as containers.
<br><br>
Regarding the available card sizes, please note that:

> - **In the Orange Design System**, a card component is available in **3 sizes** that **can not be modified**. Depending on the size of your screen, you can adjust the cards layout by changing the cards size (choosing among the 3 existing sizes) and/or the number of cards to display on a single line.

Reviewer: Is that true ? Or do we need to wait for designer to be moreprecise on this ?

Author: Dev + design told me to write this...

Reviewer: Well I don't know what to do with these informations but let's keep it for the lead dev review 🤷

- **In Boosted**, as some card variants can be used as containers, more sizing possibilities are offered to adjust their sizes.

> {{< /design-callout-alert >}}

Reviewer: The layout seems a bit weird to me atm.

Author: I kind of agree, but it not obviously better without the free lines between paragraphs. I'll try to find another way to display that info.

</div>
{{< /example >}}

{{< design-callout-alert >}}
This variant with its **centered layout** should not be used because it does not respect the Orange Design System specifications.

> Please refer to the [Cards](https://system.design.orange.com/0c1af118d/p/82eaf1-cards/b/69655e) guidelines on the Orange Design System website.

Reviewer: On the whole page
```suggestion
Please refer to the [Cards guidelines](https://system.design.orange.com/0c1af118d/p/82eaf1-cards/b/69655e) on the Orange Design System website.
```

{{< /design-callout-alert >}}

> 

Reviewer: Remove one blank line

Add some navigation to a card's header (or block) with Boosted's [nav components]({{< docsref "/components/navs-tabs" >}}).

{{< design-callout-alert >}}

> This variant should not be used because it does not respect the Orange Design System specifications.

Reviewer: So I guess the second variant is fine ? 😅

Author: No, it's not. I'll put the message in plural.

Turn an image into a card background and overlay your card's text. Depending on the image, you may or may not need additional styles or utilities.

{{< design-callout-alert >}}

> This variant should not be used because it does not respect the Orange Design System specifications. It does not respect accessibility requirements neither.

Reviewer: ```suggestion
This variant should not be used because it does not respect the Orange Design System specifications. It might lead to accessibility issues as well. Prefer using a card with a top image inside.
```

Please refer to the [Cards](https://system.design.orange.com/0c1af118d/p/82eaf1-cards/b/69655e) guidelines on the Orange Design System website.
{{< /design-callout-alert >}}

> 

Reviewer: Blank line to remove

Set a `background-color` with contrasting foreground `color` with [our `.text-bg-{color}` helpers]({{< docsref "helpers/color-background" >}}). Previously it was required to manually pair your choice of [`.text-{color}`]({{< docsref "/utilities/colors" >}}) and [`.bg-{color}`]({{< docsref "/utilities/background" >}}) utilities for styling, which you still may use if you prefer.

{{< design-callout-alert >}}

> These variants **with a colored background** should not be used because they do not respect the Orange Design System specifications.

Reviewer: ```suggestion
These variants **with a colored background** should not be used because they do not respect the Orange Design System specifications. The background should either be black or white.
```
Values might need an update

Reviewer 2: I propose:
The variants **with a colored background** (except black background) should not be used because they do not respect the Orange Design System specifications.

Use card groups to render cards as a single, attached element with equal width and height columns. Card groups start off stacked and use `display: flex;` to become attached with uniform dimensions starting at the `sm` breakpoint.

{{< design-callout-alert >}}

> These variants **with no spacing between cards** should not be used because they do not respect the Orange Design System specifications.

Reviewer: ```suggestion
These variants **with no spacing between cards** should not be used because they do not respect the Orange Design System specifications. Prefer to have `20px` between cards at least.
```
The value needs to be adapted to design needs.
